python_modules/dagster/dagster/_core/definitions/resource_annotation.py

```diff
@@ -20,12 +21,28 @@ def get_resource_args(fn) -> Sequence[Parameter]:
 RESOURCE_PARAM_METADATA = "resource_param"
 
 
+class UsableAsResourceParam(ABC):
+    """Marker class for types that can be used as a parameter on an annotated
+    function like `@asset`. Any type marked with this class does not require
+    a ResourceParam when used on an asset.
+
+    Example:
+        class YourClass(UsableAsResourceParam):
+            ...
+
+        @asset
+        def an_asset(your_class: YourClass):
+            ...
+    """
+
+
 def _is_resource_annotation(annotation: Optional[Type[Any]]) -> bool:
     from dagster._config.pythonic_config import ConfigurableResourceFactory
 
     if isinstance(annotation, type) and (
         is_subclass(annotation, ResourceDefinition)
         or is_subclass(annotation, ConfigurableResourceFactory)
```

Comment:
Should we have `ConfigurableResourceFactory` inherit this marker class?

Reply:
Yes we could do that. I wanted to be conservative and purely additive here, but I think it would be reasonable to consolidate this logic in a follow on diff.